### ISSUE TITLE
RubberBandBox leaves trails as mouse is moved.

### ISSUE THREAD

**Issue by TonyFlury** [wxWidgets/Phoenix#1824]
**Operating system**:                         Ubuntu 19.10
**wxPython version & source**:          pypi - wxPython 4.10
**Python version & source**:              stock

**Description of the problem**:
Using RubberBandBox by default leaves trails of old rectangles as the Mouse  is Dragged.

Problem can be clearly demonstrated running the Phoenix sample : Phoenix/samples/floatcanvas/DrawRect.py

Is this the intended behavior ? It seems to me from the code that it does make an attempt to draw over (in XOR mode which should erase the previous rectangle) but for whatever reason that isn't working on Ubuntu 19.10

**Comment by RobinD42** [comment]
No, it's definitely not intended. You may want to create a ticket for this at https://trac.wxwidgets.org/, if one doesn't already exist, and add a link to the ticket here so we can track it. 

Keep in mind that it could be a casualty of modernization. Most of the logical functions were the first thing to go on OSX when wx switched from QuickDraw to CoreGraphics. I don't know a lot of details of this part of the wxWidgets implementation but it's possible something similar is happening with gtk3 and cairo. (New dogs not able to do the old tricks...)

In either case, however, it would probably be a good idea to switch the `RubberBandBox` to use `wx.Overlay` instead of the XOR trick. PRs are welcome.

**Comment by TonyFlury** [comment]
Robin, I am not even sure what to raise against wxwidgets - is that XOR mode doesn't work as expected ?

**Comment by RobinD42** [comment]
Yes, XOR on a wxClientDC in the gtk3 port.  This ticket: http://trac.wxwidgets.org/ticket/16890 about working around wxINVERT no longer working also touches on the XOR issue you are seeing. So you may just want to add a comment there to give another data point.

FYI, here is a small sample to experiment with if anybody is interested.

```
import wx
print(wx.version())
print(wx.PlatformInfo)

class MyPanel(wx.Panel):
    def __init__(self, parent):
        wx.Panel.__init__(self, parent)
        self.SetBackgroundColour('light blue')
        self.timer = wx.Timer(self)
        self.timer.Start(750)
        self.Bind(wx.EVT_TIMER, self.OnTimer, self.timer)

    def OnTimer(self, evt):
        print('OnTime')
        dc = wx.ClientDC(self)
        dc.SetLogicalFunction(wx.XOR)
        dc.SetPen(wx.Pen(wx.WHITE, 4))
        dc.DrawLine(10,10, 90,90)

app = wx.App()
frm = wx.Frame(None)
pnl = MyPanel(frm)
frm.Show()
app.MainLoop()
```

**Comment by ChrisBarker-NOAA** [comment]
NOTE: wxClientDC was pretty broken on the Mac in recent versions (or didn't work at all) -- so floatcanvas.RubbberBandBox has been updated to use wxOverlay instead in this PR: #2449 

I think this can be closed.
